presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1372,6 +1316,88 @@ public AnalyzePropertyManager getAnalyzePropertyManager()
         return analyzePropertyManager;
     }
 
+    @Override
+    public MetadataResolver getMetadataResolver(Session session)
```

Comment:
New issue: the session doesn't seem to be reused and getMetadataResolver doesn't do any sort of caching. It just calls a constructor. As near as I can make out MetadataResolver is always a single use object. That suggests this really shouldn't be a separate type at all.

Even if it needs to be a separate type, session should not be an (implicit) field. Rather it should be passed to the individual methods as it was before this PR rather than used in a builder. 

Also note that *get*MetadataResolver implies this is a getter that returns the same object. *make*MetadataResolver would be slightly more idiomatic.